BDArmory/Control/MissileFire.cs

```diff
@@ -5954,7 +6024,7 @@ bool SmartPickWeapon_EngagementEnvelope(TargetInfo target)
                             case (WeaponClasses.SLW):
                                 {
                                     MissileLauncher SLW = item.Current as MissileLauncher;
-                                    if (SLW.TargetingMode == MissileBase.TargetingModes.Radar && (!_radarsEnabled && !SLW.radarLOAL)) continue; //dont select RH missiles when no radar aboard
+                                    if (SLW.TargetingMode == MissileBase.TargetingModes.Radar && (!(_sonarsEnabled || _radarsEnabled) && !SLW.radarLOAL)) continue; //dont select RH missiles when no radar aboard
                                     if (SLW.TargetingMode == MissileBase.TargetingModes.Laser && targetingPods.Count <= 0) continue; //don't select LH missiles when no FLIR aboard
```

Comment:
should be just _sonarsEnabled, so a surface radar doesn't permit torp launches; radar doesn't work underwater.